src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1974,12 +1974,20 @@ void ensureSpeculativeNodeBound()
             void bindAndRewrite()
             {
                 var remappedSymbols = _parentRemappedSymbolsOpt;
-                boundRoot = RewriteNullableBoundNodesWithSnapshots(boundRoot, binder, diagnostics, takeSnapshots: true, out var snapshotManager, ref remappedSymbols);
+                boundRoot = RewriteNullableBoundNodesWithSnapshots(boundRoot, binder, diagnostics, takeSnapshots: true, out var snapshotManager, ref remappedSymbols, shouldAvoidRewrite(Compilation));
 #if DEBUG
                 // Don't actually cache the results if the nullable analysis is not enabled in debug mode.
                 if (!Compilation.NullableSemanticAnalysisEnabled) return;
 #endif
                 GuardedAddBoundTreeForStandaloneSyntax(bindableRoot, boundRoot, snapshotManager, remappedSymbols);
+
+                static bool shouldAvoidRewrite(CSharpCompilation comp) =>
+#if DEBUG
+                    !comp.NullableSemanticAnalysisEnabled;
```

Comment:
>!comp.NullableSemanticAnalysisEnabled; [](start = 20, length = 38)

Doesn't this disable a good chunk of testing we were trying to do in DEBUG?

Reply:
I've put this comment in line with the changed source and responded there.

---
In reply to: [402667164](https://github.com/dotnet/roslyn/pull/42403#discussion_r402667164) [](ancestors = 402667164)